### PR TITLE
Bugfix/ls24003187/like variable from copy

### PR DESCRIPTION
**Pull request by davidepalladino-apuliasoft**
## Description
This work resolves the data reference of procedure (in `D` spec) to a inline param of a subroutine declared into the same procedure, by a `/COPY`. To achieve this goal, `findSize` and `findType` of `compile_time_interpreter` have to [search both in global and local scope](https://www.ibm.com/docs/en/rdfi/9.6.0?topic=considerations-scope-definitions#hscope).

Related to #LS24003187

## Checklist:
- [x] There are tests regarding this feature
- [x] The code follows the Kotlin conventions (run `./gradlew ktlintCheck`)
- [x] The code passes all tests (run `./gradlew check`)
- [ ] There is a specific documentation in the `docs` directory

### CODE REVIEW

**Comment by davidepalladino-apuliasoft** [comment]
Close for rework

**Comment by davidepalladino-apuliasoft** [comment]
Reopened after improvement.

File: rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/interpreter/compile_time_interpreter.kt [hunk 1]
```diff
@@ -309,4 +310,18 @@ open class BaseCompileTimeInterpreter(
     private fun Parm_fixedContext.findType(conf: ToAstConfiguration): Type? {
         return this.toAst(conf, emptyList()).type
     }
+
+    private fun RpgParser.RContext.getStatements(procedureName: String?): List<StatementContext> {
```
Comment (lanarimarco) [inline rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/interpreter/compile_time_interpreter.kt:314]: This function returns the statements in the following order:
1. The statements defined in a program
2. The statements defined in the program's subroutines
3. The statements defined in the procedure's subroutines
4. The statements defined in the procedure

I don't know if this is right or wrong, but if the goal is to give priority to the statements declared inside the procedure, maybe the order is not correct.

**Review by lanarimarco — COMMENTED** [review]
I had some doubts about the choice to pass the procedure's name through the CompileTimeInterpreter functions, but after spending a little time understanding your work, I realized that yours has been a good choice. 

I still have a doubt regarding the implementation of RContext.getStatements, for which I need clarification.

File: rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/interpreter/compile_time_interpreter.kt (hunk 1 above)
Comment (davidepalladino-apuliasoft) [reply rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/interpreter/compile_time_interpreter.kt:314]: I agree with you: the order isn't correct. I found a [detailed page](https://www.ibm.com/docs/en/rdfi/9.6.0?topic=definition-scope-definitions#hsbscp) which says: "_If a local item is defined with the same name as a global data item, then any references to that name inside the subprocedure use the local definition._"

**Review by lanarimarco — APPROVED** [review]
Ok

**Pull request merged**